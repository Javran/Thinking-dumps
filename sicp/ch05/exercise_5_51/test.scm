(define (map f xs)
  (if (null? xs)
      '()
      (cons (f (car xs))
            (map f (cdr xs)))))

(define (add1 x)
  (+ x 1))

(display (map add1 '(1 2 3 4 5)))
(newline)

(map
 (lambda (x)
   (display x)(newline))
 '("this" "is" "a" "test"))
