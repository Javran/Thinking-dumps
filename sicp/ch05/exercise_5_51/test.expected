(2 3 4 5 6)
(1 . 2)
this
is
a
test
91
